From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Eikum <aeikum@codeweavers.com>
Date: Fri, 30 Mar 2018 10:40:43 -0500
Subject: [PATCH] HACK: winedbg: When crash dialog is not shown, dump crash
 info to stderr

Use the real unix stderr via __wine_dbg_output().

CW-Bug-Id: #21872
Valve-commit: 12e7d48882a3b61db6432fe393a8972157207e51
Picked: resolved
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 programs/winedbg/debugger.h   | 1 +
 programs/winedbg/tgt_active.c | 3 +++
 programs/winedbg/winedbg.c    | 7 +++++++
 3 files changed, 11 insertions(+)

diff --git a/programs/winedbg/debugger.h b/programs/winedbg/debugger.h
index 3ceef9c..7c5493f 100644
--- a/programs/winedbg/debugger.h
+++ b/programs/winedbg/debugger.h
@@ -313,6 +313,7 @@ extern  dbg_ctx_t               dbg_context;
 extern  BOOL                    dbg_interactiveP;
 extern  HANDLE                  dbg_houtput;
 extern  struct list             dbg_process_list;
+extern  BOOL                    dbg_use_wine_dbg_output;
 
 struct dbg_internal_var
 {
diff --git a/programs/winedbg/tgt_active.c b/programs/winedbg/tgt_active.c
index f3041e3..5247cf3 100644
--- a/programs/winedbg/tgt_active.c
+++ b/programs/winedbg/tgt_active.c
@@ -938,6 +938,9 @@ enum dbg_start dbg_active_auto(int argc, char* argv[])
         if (event) thread = display_crash_details( event );
         if (thread) dbg_houtput = output = create_temp_file();
         break;
+    case TRUE:
+        dbg_use_wine_dbg_output = TRUE;
+        break;
     }
 
     input = parser_generate_command_file("echo Modules:", "info share",
diff --git a/programs/winedbg/winedbg.c b/programs/winedbg/winedbg.c
index 9707120..b4fa402 100644
--- a/programs/winedbg/winedbg.c
+++ b/programs/winedbg/winedbg.c
@@ -82,6 +82,7 @@ DWORD	                dbg_curr_pid = 0;
 dbg_ctx_t               dbg_context;
 BOOL    	        dbg_interactiveP = FALSE;
 HANDLE                  dbg_houtput = 0;
+BOOL                    dbg_use_wine_dbg_output = FALSE;
 struct list             dbg_process_list = LIST_INIT(dbg_process_list);
 
 struct dbg_internal_var         dbg_internal_vars[DBG_IV_LAST];
@@ -93,6 +94,12 @@ static void dbg_outputA(const char* buffer, int len)
 
     DWORD w, i;
 
+    if (dbg_use_wine_dbg_output)
+    {
+        __wine_dbg_output(buffer);
+        return;
+    }
+
     while (len > 0)
     {
         unsigned int count = min( len, sizeof(line_buff) - line_pos );
